pkg/cni/pool/pool.go

```diff
@@ -0,0 +1,67 @@
+/*
+ Copyright 2023, NVIDIA CORPORATION & AFFILIATES
+ Licensed under the Apache License, Version 2.0 (the "License");
+ you may not use this file except in compliance with the License.
+ You may obtain a copy of the License at
+     http://www.apache.org/licenses/LICENSE-2.0
+ Unless required by applicable law or agreed to in writing, software
+ distributed under the License is distributed on an "AS IS" BASIS,
+ WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ See the License for the specific language governing permissions and
+ limitations under the License.
+*/
+
+package pool
+
+import (
+	"encoding/json"
+	"fmt"
+
+	v1 "k8s.io/api/core/v1"
+)
+
+const (
+	ipBlocksAnnotation = "ipam.nvidia.com/ip-blocks"
+)
+
+type IPPoolConfig struct {
+	Subnet  string `json:"subnet"`
+	StartIP string `json:"startIP"`
+	EndIP   string `json:"endIP"`
+	Gateway string `json:"gateway"`
+}
+
+type Manager interface {
+	// GetPoolByName returns IPPoolConfig for the provided pool name or nil if pool doesnt exist
+	GetPoolByName(name string) *IPPoolConfig
+}
+
+type ManagerImpl struct {
+	poolByName map[string]*IPPoolConfig
+}
+
+func NewManagerImpl(node *v1.Node) (*ManagerImpl, error) {
```

Comment:
nit: rename to New?

Reply:
Personally i prefer to return concrete type from a Ctor method and let caller assign to an interface.

both appraches are correct imo, just need to stick to one. LMK if there are special concern that require me to change it.

Reply:
let's keep it as is